### PR TITLE
Bug fix/8395 java pom integrated legacy action converter

### PR DESCRIPTION
**Pull request by ravirk91**
Bug fix/8395 java pom integrated legacy action converter

### CODE REVIEW

File: Ginger/GingerTest/ActionConversion/ActionConversionTest.cs [hunk 1]
```diff
@@ -991,6 +1004,29 @@ public void JavaActSwitchWindowToUISwitchWindow()
...
+        [TestMethod]
+        public void JavaLegacyPOMConversionTest()
+        {
+            //Arrange
+            NewRepositorySerializer RepositorySerializer = new NewRepositorySerializer();
+            var businessFlowFile = TestResources.GetTestResourcesFile(@"JavaLegacyToPOMxml" + Path.DirectorySeparatorChar + "Java_Legacy_Actions_BF.Ginger.BusinessFlow.xml");
+            var javaPomFile = TestResources.GetTestResourcesFile(@"JavaLegacyToPOMxml\Java Swing Test App.Ginger.ApplicationPOMModel.xml");
+            
+            //Load BF
+            mBF = (BusinessFlow)RepositorySerializer.DeserializeFromFile(businessFlowFile);
```
Comment (ambarkhan) [inline Ginger/GingerTest/ActionConversion/ActionConversionTest.cs]: Codacy Issue found: [Make the enclosing instance method 'static' or remove this set on the 'static' field.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4824688)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4824688&bid=16035582)

File: Ginger/GingerTest/ActionConversion/ActionConversionTest.cs (hunk 1 above)
Comment (ambarkhan) [inline Ginger/GingerTest/ActionConversion/ActionConversionTest.cs]: Codacy Issue found: [Make the enclosing instance method 'static' or remove this set on the 'static' field.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4824688)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4824688&bid=16035582)

File: Ginger/GingerTest/ActionConversion/ActionConversionTest.cs [hunk 2]
```diff
@@ -682,7 +695,7 @@ public void JavaActGenericSetValueToUIElementSetValue()
         }
 
         [TestMethod]
-        [Timeout(60000)]
+       // [Timeout(60000)]
```
Comment (JinendraGhodke) [inline Ginger/GingerTest/ActionConversion/ActionConversionTest.cs]: uncomment this

File: Ginger/GingerTest/ActionConversion/ActionConversionTest.cs [hunk 3]
```diff
@@ -991,6 +1004,29 @@ public void JavaActSwitchWindowToUISwitchWindow()
             ActSwitchWindowTOUISwitchWindowConvertor(activity);
         }
 
+        [TestMethod]
+        public void JavaLegacyPOMConversionTest()
```
Comment (JinendraGhodke) [inline Ginger/GingerTest/ActionConversion/ActionConversionTest.cs]: add timeout

File: Ginger/GingerTest/ActionConversion/ActionConversionTest.cs [hunk 4]
```diff
@@ -991,6 +1004,29 @@ public void JavaActSwitchWindowToUISwitchWindow()
...
+
+            mBF.Activities[0].SelectedForConversion = true;
+
+            //Load POM
+            ApplicationPOMModel applicationPOM = (ApplicationPOMModel)RepositorySerializer.DeserializeFromFile(javaPomFile);
+            mSolutionRepository.AddRepositoryItem(applicationPOM);
+
+            ExecuteActionConversion(true, true, string.Empty, true, applicationPOM.Guid);
+            //Assert
+            Assert.AreEqual(((ActUIElement)mBF.Activities[1].Acts[1]).ElementLocateBy.ToString(),eLocateBy.POMElement.ToString());
```
Comment (JinendraGhodke) [inline Ginger/GingerTest/ActionConversion/ActionConversionTest.cs]: validation Element action as well

File: Ginger/GingerTest/ActionConversion/ActionConversionTest.cs (hunk 2 above)
Comment (ravirk91) [reply Ginger/GingerTest/ActionConversion/ActionConversionTest.cs]: changes done.

File: Ginger/GingerTest/ActionConversion/ActionConversionTest.cs (hunk 4 above)
Comment (ravirk91) [reply Ginger/GingerTest/ActionConversion/ActionConversionTest.cs]: Added validation for action.

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4824688&bid=16035582)

File: Ginger/GingerTest/ActionConversion/ActionConversionTest.cs [hunk 5]
```diff
@@ -991,6 +1004,30 @@ public void JavaActSwitchWindowToUISwitchWindow()
...
+        [Timeout(100000)]
+        public void JavaLegacyPOMConversionTest()
+        {
+            //Arrange
+            NewRepositorySerializer RepositorySerializer = new NewRepositorySerializer();
+            var businessFlowFile = TestResources.GetTestResourcesFile(@"JavaLegacyToPOMxml" + Path.DirectorySeparatorChar + "Java_Legacy_Actions_BF.Ginger.BusinessFlow.xml");
+            var javaPomFile = TestResources.GetTestResourcesFile(@"JavaLegacyToPOMxml\Java Swing Test App.Ginger.ApplicationPOMModel.xml");
+            
+            //Load BF
+            mBF = (BusinessFlow)RepositorySerializer.DeserializeFromFile(businessFlowFile);
```
Comment (ambarkhan) [inline Ginger/GingerTest/ActionConversion/ActionConversionTest.cs]: Codacy Issue found: [Make the enclosing instance method 'static' or remove this set on the 'static' field.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4824688)

**Pull request merged**